lib/backend/model/bgppeer.go

```diff
@@ -185,3 +195,18 @@ type BGPPeer struct {
 	// configuration.
 	ASNum numorstring.ASNumber `json:"as_num,string"`
 }
+
+func extractIPAndPort(ipPort string) ([]byte, string) {
+	arr := strings.Split(ipPort, ipPortSeparator)
+	if len(arr) == 2 {
+		return []byte(arr[0]), arr[1]
+	}
+	return []byte(ipPort), ""
+}
+
+func combineIPAndPort(ip string, port string) string {
```

Comment:
I think you should map port 179 (the default BGP port) to the old style of key.  That will ensure that you can't have an old-style peer and a new-style peer both with the same port.  It'll also mean that old typha clients and new typhas will interop more cleanly.

Reply:
We are not setting default port to 179 explicitly anywhere, so I think if user explicitly sets 179 in new-style there will be a conflict.

Reply:
so we will always get the old-style peer key if port is not explicit set.

Reply:
How about doing these translations:
```
BGPPeerKey{..., Port: 179} -> "/calico/bgp/v1/.../peer_v4/10.0.0.5"
"/calico/bgp/v1/.../peer_v4/10.0.0.5" -> BGPPeerKey{..., Port: 179} 
```
I.e. never use 179 in the string version of the key but fill it in on read so that it's explicit for new clients.

I think we should avoid allowing an old and new-style version of hte 179 peer being created; presumably that would cause BIRD to fail and effectively break the cluster?

Reply:
Thanks for clarifying @fasaxc, you're right, having an old peer and a new peer with port 179 will cause issue. Not using 179 in key will help with that :+1: